odl/trafos/wavelet.py

```diff
@@ -344,36 +217,43 @@ def pywt_coeff_to_array3d(coeff, size_list, nscales):
         Flattened and concatenated coefficient array
         The length of the array depends on the size of input image to
         be transformed and on the chosen wavelet basis.
-        If the size of the input image is 2^n x 2^n, the length of the
-        wavelet coefficient array is the same.
-    """
-    size_flatCoeff = np.prod(size_list[0])
-    for kk in range(1, nscales + 1):
-        size_flatCoeff += 7 * np.prod(size_list[kk])
+      """
+    flat_sizes = [np.prod(shp) for shp in size_list[:-1]]
+    if np.size(size_list[0]) == 1:
+        const = 1
+    if np.size(size_list[0]) == 2:
+        const = 3
+    elif np.size(size_list[0]) == 3:
```

Comment:
Would something of the sort
`else:
        raise NotImplementedError('ndim {} not 1, 2 or 3'
                                  ''.format(len(size_list[0])))`

do? Since only 1d, 2d and 3d are implemented...